pkgs/crypto/lib/src/hash_sink.dart

```diff
@@ -79,18 +78,48 @@ abstract class HashSink implements Sink<List<int>> {
   void add(List<int> data) {
     if (_isClosed) throw StateError('Hash.add() called after close().');
     _lengthInBytes += data.length;
-    _pendingData.addAll(data);
-    _iterate();
+    _addData(data);
+  }
+
+  void _addData(List<int> data) {
+    var dataIdx = 0;
+    var currentChunkNextIndex = _currentChunkNextIndex;
+    final size = _currentChunk.length;
+    _byteDataView ??= _currentChunk.buffer.asByteData();
+    while (true) {
+      // Is there enough data left in [data] for a full chunk?
+      var restEnd = currentChunkNextIndex + data.length - dataIdx;
+      if (restEnd < size) {
+        // No. Just add into [_currentChunk].
+        _currentChunk.setRange(currentChunkNextIndex, restEnd, data, dataIdx);
+        _currentChunkNextIndex = restEnd;
+        return;
+      }
+
+      // Yes. Fill out [_currentChunk] and process it.
+      _currentChunk.setRange(currentChunkNextIndex, size, data, dataIdx);
+      dataIdx += size - currentChunkNextIndex;
+
+      // Now do endian conversion to words.
+      var j = 0;
+      do {
+        _currentChunk32[j] =
+            _byteDataView!.getUint32(j * bytesPerWord, _endian);
+        j++;
+      } while (j < _currentChunk32.length);
```

Comment:
```suggestion
for (var j = 0; j < _currentChunk32.length; j++) {
  _currentChunk32[j] = _byteDataView!.getUint32(j * bytesPerWord, _endian);
}
```

Reply:
For at little more detail, doing this change results in this:

```
msec task-clock:u: 6.5173% +/- 0.5320% (48.11 +/- 3.93) (738.26 -> 786.37)
cycles:u: 6.9092% +/- 0.5230% (211044994.56 +/- 15976750.57) (3054537912.36 -> 3265582906.92)
instructions:u: 3.4120% +/- 0.0000% (378546773.16 +/- 2888.68) (11094594602.64 -> 11473141375.80)
seconds time elapsed: 6.5162% +/- 0.5333% (0.05 +/- 0.00) (0.74 -> 0.79)
seconds user: 6.9181% +/- 0.6133% (0.05 +/- 0.00) (0.70 -> 0.74)
```

i.e. it's 3.412% more instructions executed and takes ~6.5% longer when running on the `wasm-opt` input file I also used in the statistics on the PR.